disnake/ext/commands/cog.py

```diff
@@ -428,7 +434,13 @@ def decorator(func: FuncT) -> FuncT:
             if not asyncio.iscoroutinefunction(actual):
                 raise TypeError("Listener function must be a coroutine function.")
             actual.__cog_listener__ = True
-            to_assign = name or actual.__name__
+            to_assign = (
+                name
+                if isinstance(name, str)
+                else "on_" + name.value
+                if issubclass(name.__class__, _EnumValueBase)
+                else actual.__name__
+            )
```

Comment:
```suggestion
            to_assign = (
                actual.__name__
                if name is MISSING
                else (name if isinstance(name, str) else f"on_{name.value}")
            )
```
I feel like this is a bit more readable, nested conditionals are somewhat difficult to parse